src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -163,7 +179,9 @@ long getThrottlingWaitTimeMicros() {
             synchronized (this) {
                 throwIfNoConnection();
                 return connection.unsubscribe(topic).thenApply((i) -> {
-                    subscriptionTopics.remove(topic);
+                    synchronized (this) {
```

Comment:
this is fine, but I don't think it is actually necessary because I really only want to synchronize the pair of the map and inprogress.